src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1027,7 +1024,85 @@ public async Task<NuGetProject> UpgradeProjectToPackageReferenceAsync(NuGetProje
 
             return nuGetProject;
         }
+        #endregion
+
+        #region IVsSolutionEvents
+        public int OnAfterOpenProject(IVsHierarchy pHierarchy, int fAdded)
```

Comment:
Is this equivalent to project added?

Reply:
If so, can we clean up the old events eventually? (Don't bother now, now the priority is to make things just work).

Reply:
[doc](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivssolutionevents.onafteropenproject?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_Interop_IVsSolutionEvents_OnAfterOpenProject_Microsoft_VisualStudio_Shell_Interop_IVsHierarchy_System_Int32_) says, `Notifies listening clients that the project has been opened`

Reply:
yes, OnAfterOpenProject is equivalent to DTE.SolutionEvents.ProjectAdded. It's just we have an IVsHierarchy, rather than a DTE.Project. But I know how we can get a DTE.Project from a IVsHeirarchy.

I would like to "clean up" and move away from DTE and towards IVs APIs (I checked the DTE source and it uses all these IVs* APIs under the covers, so DTE might have been more convenient to use, but it's an overhead making NuGet slower).  However, the perf benefit I expect is so minimal that it's not justifyable from a data driven decision making point of view.

Reply:
>  However, the perf benefit I expect is so minimal that it's not justifyable from a data driven decision making point of view.

I'd say only do it if you can clean it up in less than 1-2 hours :)